core/src/main/java/greencity/controller/UserController.java

```diff
@@ -59,20 +66,26 @@ public class UserController {
      */
     @Operation(summary = "Update status of user")
     @ApiResponses(value = {
-        @ApiResponse(responseCode = "200", description = HttpStatuses.OK,
-            content = @Content(schema = @Schema(implementation = UserStatus.class))),
-        @ApiResponse(responseCode = "303", description = HttpStatuses.SEE_OTHER),
-        @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST),
-        @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED),
-        @ApiResponse(responseCode = "403", description = HttpStatuses.FORBIDDEN)
+            @ApiResponse(responseCode = "200", description = HttpStatuses.OK,
+                    content = @Content(schema = @Schema(implementation = UserStatus.class))),
+            @ApiResponse(responseCode = "303", description = HttpStatuses.SEE_OTHER),
+            @ApiResponse(responseCode = "400", description = HttpStatuses.BAD_REQUEST),
+            @ApiResponse(responseCode = "401", description = HttpStatuses.UNAUTHORIZED),
+            @ApiResponse(responseCode = "403", description = HttpStatuses.FORBIDDEN)
     })
     @PatchMapping("status")
     public ResponseEntity<UserStatusDto> updateStatus(
-        @Valid @RequestBody UserStatusDto userStatusDto, @ApiIgnore Principal principal) {
-        return ResponseEntity.status(HttpStatus.OK)
-            .body(
-                userService.updateStatus(
-                    userStatusDto.getId(), userStatusDto.getUserStatus(), principal.getName()));
+            @Valid @RequestBody UserStatusDto userStatusDto, @ApiIgnore Principal principal) {
+        if (principal == null) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).build();
+        }
+        try {
+            UserStatusDto updatedStatus = userService.updateStatus(
+                    userStatusDto.getId(), userStatusDto.getUserStatus(), principal.getName());
+            return ResponseEntity.status(HttpStatus.OK).body(updatedStatus);
+        } catch (BadUpdateRequestException | LowRoleLevelException e) {
+            return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
+        }
```

Comment:
Why did you modify this method? Task 289 about the bug in updateUserProfilePicture method?